src/frontend/src/pages/ProjectPage/pages/ContractPage/pages/ManagePersonnelPage/components/ExcelImportSideSheet/index.tsx

```diff
@@ -0,0 +1,137 @@
+    isOpen: boolean;
+    setIsOpen: React.Dispatch<React.SetStateAction<boolean>>;
+    processingError: boolean;
+};
+
+const ExcelImportSideSheet: React.FC<ExcelImportSideSheetProps> = ({
+    setSelectedFile,
+    isProccessing,
+    isOpen,
+    setIsOpen,
+    processingError,
+}) => {
+    const [selectedFileForUpload, setSelectedFileForUpload] = React.useState<File | null>(null);
+    const fileInput = React.useRef<HTMLInputElement>(null);
+    const [fileError, setFileError] = React.useState<string | null>(null);
+
+    React.useEffect(() => {
+        setSelectedFileForUpload(null);
+        setFileError(null);
+    }, [isOpen]);
+
+    const validateAndSetFile = React.useCallback((file: File) => {
+        const fileExtension = file.name.substr(file.name.lastIndexOf('.') + 1).toLowerCase();
+
+        if (fileExtension !== 'xlsx') {
+            setFileError('Fileformat not supported. Please again with a xlsx file');
+            setSelectedFileForUpload(null);
+            return;
+        }
+        setFileError(null);
+        setSelectedFileForUpload(file);
+    }, []);
+
+    const dragDropFileUpload = React.useCallback((e: React.DragEvent<HTMLDivElement>) => {
+        e.preventDefault();
+        if (!e.dataTransfer.items || e.dataTransfer.items[0].kind !== 'file') return;
+
+        const file = e.dataTransfer.items[0].getAsFile();
+        if (!file) return;
+        validateAndSetFile(file);
+    }, []);
+
+    const inputFileUpload = React.useCallback((e: React.ChangeEvent<HTMLInputElement>) => {
+        e.preventDefault();
+
+        const file = e.target.files ? e.target.files[0] : null;
+        if (!file) return;
+        validateAndSetFile(file);
+    }, []);
+
+    const fileInputClick = React.useCallback(() => {
+        fileInput?.current?.click();
+    }, [fileInput]);
+
+    const startProcessingSelectedFile = React.useCallback(() => {
+        setSelectedFile(selectedFileForUpload);
+    }, [selectedFileForUpload]);
+
+    const stopPropagationAndDefault = React.useCallback((e: React.DragEvent<HTMLDivElement>) => {
+        e.stopPropagation();
+        e.preventDefault();
+    }, []);
+
+    const closeSidesheet = React.useCallback(() => {
+        !isProccessing && setIsOpen(false);
+    }, [isProccessing]);
+
+    return (
+        <ModalSideSheet
+            header="Excel Import"
+            show={isOpen}
+            size={'medium'}
+            onClose={closeSidesheet}
+        >
+            <div className={styles.container}>
+                <div onClick={(e) => e.stopPropagation()}>
+                    {isProccessing && <Spinner centered title="Processing Excel file" />}
+                    {!isProccessing && (
+                        <>
+                            <div
+                                className={styles.dragDropContainer}
+                                onDrop={dragDropFileUpload}
+                                onDragEnter={(e) => stopPropagationAndDefault(e)}
```

Comment:
Instead of defining a new callback function here, could we just use the one defined above. Eg. `onDragEnter={stopPropagationAndDefault}`